### DISCUSSION TITLE
Enable logging of lengthly operations (duration)

### DISCUSSION THREAD

**Discussion by ssbarnea** [Textualize/rich#894] [Ideas] [answered]
As it is not uncommon to want to log duration of actions that are likely to be time consuming it would be really nice if rich could provide some extensions that enable users to do so.

The usually approach is to record the start time and include the computed duration after the action finished. Optionally some users decide to log both the start and the end of the action, but this makes the logging extra verbose. 

If we would be able to have a decorator or a context manager that we can easily use to implement this kind of logging it would be great.

**Comment by willmcgugan** [comment, marked as answer]
Seems beyond the remit of Rich. There may be packages you could adapt to write to Rich, or roll your own in a dozen lines or so.

**Reply by ssbarnea** [reply]
Sure. But I had to ask. I will post a link when I pick one solution, it may help someone.
